doofinder-for-woocommerce/assets/js/wizard.js

```diff
@@ -104,15 +58,65 @@ jQuery(() => {
 
   $(".open-window").click((e) => {
     const pageType = $(e.currentTarget).attr("data-type");
-
     connectModal = popupCenter({
-      url: `https://app.doofinder.com/plugins/${pageType}/woocommerce?email=${doofinderConnectEmail}&token=${doofinderConnectToken}&return_path=${doofinderConnectReturnPath}`,
+      url: `${doofinderAdminPath}/plugins/${pageType}/woocommerce?email=${doofinderConnectEmail}&token=${doofinderConnectToken}`,
       title: "DoofinderConnect",
       w: 600,
       h: 700,
     });
-
-    clearInterval(checkInterval);
-    checkInterval = setInterval(ajaxCheckData, 1000);
   });
+
+  window.addEventListener(
+    "message",
+    (event) => {
+      //Check that the sender is doofinder
+      if (event.origin !== doofinderAdminPath) return;

```

Comment:
Could we habe trouble with the middleware?

Reply:
Good question, I'll try to make some checks

Reply:
I've added a regular expression to check if the domain is from doofinder